Reviews check

aOrro opened the pull request:
Hola Dani, lo primero, perdona la desorganización de mi CSS.

He añadido algunas imágenes para darle un poco de vidilla, pero como puedes ver, en la sección de reviews, me ha creado un espacio enorme entre esa sección y la de instagram. Podrías echarle un vistazo y decirme cómo solucionarlo?

Estoy planteándome seriamente el empezar de cero todo el CSS... Que piensas?

aOrro committed:
Update index.html

aOrro committed:
Update styles.css

delacruz-dev commented on index.html:
```diff
@@ -159,23 +174,34 @@ <h3>The damage</h3>
+            <h3>Opening hours</h3>
+            <div class="dash-detail"></div>
+            <div class="timetable">
+              <span class="days">Monday - Friday:&nbsp &nbsp &nbsp &nbsp &nbsp &nbsp &nbsp &nbsp &nbsp &nbsp &nbsp &nbsp9:00 - 21:00</span>
```
Aquí puedes utilizar un `display: flex` con `space-between` para posicionar el día y la hora y ahorrarte tanto `&nbsp`.

aOrro replied:
Si, eso fue hecho a prisa el dia anterior a la presentación para dar el efecto visual, pensaba modificarlo cuando llegue a esa parte

delacruz-dev commented on index.html:
```diff
@@ -159,23 +174,34 @@ <h3>The damage</h3>
 
       <div class="copy">
+        <div class="copy-social">
+          <a href="#" class="fa fa-instagram"></a>
```
Recuerda que si coges los SVG de font awesome, te ahorrarás unos cuantos KB de cargar toda la librería y tu página cargará más rápida.

aOrro replied:
lo intenté, y te dan un <i> para pegar en tu html, que en teoría es el icono, y lo metí. La cosa en que no lo veía por ninguna parte. Los busque con devtools y estaban en la pagina pero ponía 0x0 de tamaño... eso que es, height and width? Intente cambiarlos pero no conseguí que se hiciesen mas grandes

delacruz-dev commented on styles.css:
```diff
@@ -107,42 +122,77 @@ body {
+  box-shadow: 0 2.5rem 4rem rgba(0, 0, 0, .5);
+}
+
+.about-images .about-img-2 {
```
Recuerda que puedes utilizar pseudo selectores `:nth-child`, para no ser tan específico con las imágenes.

aOrro replied:
También lo hice pero no funciona por algun motivo...

delacruz-dev replied:
La sintaxis correcta es:

```css

.about-images img:nth-child(1) {
  top: 0rem;
  left: 15rem;
}

.about-images img:nth-child(2) {
  top: 13rem;
  left: 5rem;
}

.about-images img:nth-child(3) {
  top: 19rem;
  left: 25rem;
}
```

Lo he probado en local y me funciona. Tienes la documentación en MDN: https://developer.mozilla.org/es/docs/Web/CSS/:nth-child

aOrro replied:
Perfecto, se lo estaba aplicando a la clase .about-images en vez de al elemento img.

delacruz-dev commented:
![image](https://user-images.githubusercontent.com/5173869/97774129-c10da880-1b55-11eb-802e-b6636050b41b.png)

Te falta por subir estas imágenes a git.

¿Por qué estás planteándote hacer el CSS desde cero? ¿Qué es lo que harías distinto?

Por otro lado, no sé a qué te refieres con que hay un espacio enorme entre la sección de reviews y la de instagram. A lo mejor es porque me faltan imágenes, pero lo veo "bien".

delacruz-dev commented on styles.css:
```diff
@@ -107,42 +122,77 @@ body {
+  box-shadow: 0 2.5rem 4rem rgba(0, 0, 0, .5);
 }
 
-.about-section .about-images .about-img-2 {
-  top:5rem;
-  left: 20rem;
+.about-img-2:hover {
+  z-index: 90;
+  transform: scale(1.05) translateY(-.5rem);
+  box-shadow: 0 2.5rem 4rem rgba(0, 0, 0, .5);
 }
 
-.about-section .about-images .about-img-3 {
-  top: 20rem;
-  left: 28rem;
+.about-img-3:hover {
+  z-index: 90;
+  transform: scale(1.05) translateY(-.5rem);
+  box-shadow: 0 2.5rem 4rem rgba(0, 0, 0, .5);
+}
```
Si estos tres estilos hacen exactamente lo mismo, ¿No podrías simplificarlo en una sola clase?

aOrro replied:
Pues si! lo que hice fue poner uno detrás del otro con comas pero no funcionó. Pero por supuesto, tiene mas sentido crear una clase única para ello.

Voy a dale caña y t pongo al dia luego. Muchas gracias!

aOrro replied:
He subido las imágenes que me faltaban... menos la de favicon que no se cual es..
De todas formas debería ser suficiente para ver lo del espacio que te comenté.

Yo creo que se debe a insertar la imágenes esas de apoyo al lado de la tabla de precios y la review, despues de hacerlas más pequeñas y recolocarlas (por cierto si me das un método mejor que top negativo y/o left/right seía la leche).

aOrro replied:
Bueno Dani, me he partido la cabeza un poco durante el finde y he conseguido dale algo de forma a la cosa.
Te pongo al dia:

1) Los pseudo selectores ntn-child no funcionan en las imágenes de la sección about por algún motivo que desconozco.

2) He insertado los iconos individuales desde fontawesome como me has aconsejado, pero no soy capaz de hacer que se vean, cuando utilizo devtools puedo ver que están en la página pero no consigo darles tamaño, ni siguiendo la documentacion de fontawesome...

3) He añadido la funcionalidad para hacer un slide en la seccion de reviews y funciona, pero hay un problema, no soy capaz de centar el DIV con el contenido para que quede centrado verticalmente en la sección, lo mismo pasa con las flechas laterales y con la imagen/dibujo que le meti como decoración abajo a la derecha del comentario. Podrias echarme un cable con eso?

4) En la sección de la tabla de precio también he insertado un par de imágenes/dibujo para darle vidilla y acompañar a la tabla de precios que se veia muy vacío el espacio. Pero no soy capaz de hacer que las imágenes queden a los lados alineadas con la parte de abajo de la tabla... Necesito ayuda con eso también.

5) Como puedes ver, ya he implementado el mapa de Google utilizando su API, la cosa es que me piden una billing address, supongo que tendré que pedir esto a mi cliente verdad? No quería ir muy lejos sin preguntarte primero.

En principio eso sería todo, si tienes algún consejo de sobre qué cosas limpiar en el documento CSS o de como trabajar los márgenes y espaciado entre secciones mejor, soy todo oídos.

Otra vez más, gracias por tu ayuda!

aOrro replied:
Espero ansioso tu respuesta y opinión sobre el avance.

Que tengas un buen Lunes!

delacruz-dev replied:
He dejado un par de comentarios, pero no había visto este. Luego me lo reviso con más calma.

delacruz-dev replied:
Sobre el punto 3.

No te queda centrado porque lo estás haciendo respecto al div en diagonal. Utiliza dos divs distintos, uno para hacer el efecto en diagonal, que quedará en el fondo, y otro encima de este que esté horizontal para tenerlo todo centrado.

Sobre el punto 4.

No veo las imágenes, me da error la consola al intentar obtenerlas. ¿Puedes subirlas?

Sobre el 5.

Si utilizas Google maps, no es gratuito. Un truco es poner una captura con la imagen del mapa y al hacer clic que te lleve a Google maps.

aOrro replied:
Ahora mismo haré lo del div, las imágenes están subidas ya asi que ahora podrás ver a lo que me refiero en la seccion de precios y reviews. Creo que Google maps era gratis hasta 25k visitas mensuales... pero bueno, tendré que buscar otra forma

aOrro committed:
Create app.js

Funcionalidades para la sección de reviews (efecto carousel) y el mapa de localización de Google

aOrro committed:
Update index.html

aOrro committed:
Update styles.css

delacruz-dev commented on app.js:
```diff
@@ -0,0 +1,43 @@
+  let dots = document.getElementsByClassName("dot");
+
+  if (n > slides.length) {slideIndex = 1}
+    if (n < 1) {slideIndex = slides.length}
```
La indentación es extraña. Cuando lo he visto por primera vez, pensaba que el segundo if se ejecutaba dentro del bloque del primero, pero en realidad no es así.

Te recomiendo utilizar Prettier para que el editor de código te solucione estas cosas automáticamente:

https://prettier.io/

aOrro replied:
básicamente lo copié de w3s porque no sabía como implementar la funcionalidad, pero le daré un repaso ya que solo cambié var por let y la referencia al la clase y ya

aOrro replied:
Le he dada un repaso a la identación y la he hecho como la escribiría yo, a ver que te parece

delacruz-dev commented:
![image](https://user-images.githubusercontent.com/5173869/97959952-b8fd7500-1db0-11eb-8fcd-6de55d3a8ba3.png)

Este elemento tiene un efecto curioso, pero se rompe al redimensionar la pantalla, te va a dar bastante guerra al hacerlo responsive. Quizás deberías empezar por definir cómo se tiene que ver en distintos viewports, o darle una vuelta e intentar otra composición.

delacruz-dev commented on index.html:
```diff
@@ -1,87 +1,100 @@
+      </ul>
+
+      <div class="nav-social">
+        <i class="fab fa-instagram"></i>
```
Para sustituir los iconos por SVGs, tienes que darle un width o no se verán. Mira el ejemplo, donde pongo width="20" (pon el valor que creas conveniente):

```suggestion
<svg aria-hidden="true" focusable="false" data-prefix="fab" data-icon="instagram" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" width="20"><path fill="currentColor" d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z" class=""></path></svg>
```

aOrro replied:
De donde sacaste el svg?? en mi documento solo tengo los `<i>`

aOrro replied:
en la pagina de fontawesome solo me dan el codigo que pegue en el html

delacruz-dev commented on index.html:
```diff
@@ -1,87 +1,100 @@
-          <img src="../src/images/about1.jpg" alt="" class="about-img about-img-1">
-          <img src="../src/images/about2.jpg" alt="" class="about-img about-img-2">
-          <img src="../src/images/about3.jpg" alt="" class="about-img about-img-3">
+          <div>
+            <img src="../src/images/about2.jpg" alt="" class="about-img about-img-1 effect">
+          </div>
+          <div>
+            <img src="../src/images/about1.jpg" alt="" class="about-img about-img-2 effect">
+          </div>
+          <div>
+            <img src="../src/images/about3.jpg" alt="" class="about-img about-img-3 effect">
+          </div>
```
Saca las imágenes de sus div, si no son necesarios.

aOrro committed:
Add files via upload

aOrro committed:
Add files via upload

aOrro committed:
Update app.js
